crates/primitives/src/db/components/block_hash.rs

```diff
@@ -30,3 +30,16 @@ where
         BlockHashRef::block_hash(*self, number)
     }
 }
+
+#[cfg(feature = "std")]
```

Comment:
There is `alloc::sync::Arc` but to be honest I don't know if it would make a problem and if they can be interchangeably used